### PR TITLE
Add tests for ListenIP interface/IP scoring

### PR DESCRIPTION
**Pull request by prashantv**
Some prefactoring for #554

### CODE REVIEW

File: localip.go [hunk 1]
```diff
@@ -54,18 +54,12 @@ func scoreAddr(iface net.Interface, addr net.Addr) (int, net.IP) {
...
-	if err != nil {
-		return nil, err
-	}
-
+func listenIP(interfaces []net.Interface) (net.IP, error) {
 	bestScore := -1
 	var bestIP net.IP
 	// Select the highest scoring IP as the best IP.
 	for _, iface := range interfaces {
+
```
Comment (monicato) [inline localip.go]: nit: remove newline

File: localip_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
```
Comment (witriew) [inline localip_test.go]: nit: year

File: localip_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			msg:    "non-local up ipv4 IPNet address",
+			iface:  net.Interface{Flags: net.FlagUp},
+			addr:   &net.IPNet{IP: testIP},
+			want:   500,
+			wantIP: testIP,
+		},
+		{
+			msg:    "non-local up ipv4 IPAddr address",
+			iface:  net.Interface{Flags: net.FlagUp},
+			addr:   &net.IPNet{IP: testIP},
```
Comment (witriew) [inline localip_test.go]: `&net.IPAddr`

File: localip_test.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			msg:    "non-local down ipv4 address",
+			iface:  net.Interface{},
+			addr:   &net.IPNet{IP: testIP},
+			want:   400,
+			wantIP: testIP,
+		},
+		{
+			msg:    "non-local down ipv6 address",
+			iface:  net.Interface{},
+			addr:   &net.IPAddr{IP: net.ParseIP("2001:db8:a0b:12f0::1")},
```
Comment (witriew) [inline localip_test.go]: Declare this IP at the top as well (dedup L65).

File: localip_test.go (hunk 2 above)
Comment (prashantv) [inline localip_test.go]: Keeping it 2015 to keep it consistent with the rest of the code, apparently the "year of first publication" is what we want:
http://www.contentious.com/2007/01/07/copyright-notice-is-the-year-really-necessary/

File: localip_test.go (hunk 3 above)
Comment (prashantv) [inline localip_test.go]: Nice catch, thanks

File: localip_test.go (hunk 4 above)
Comment (prashantv) [inline localip_test.go]: Done

File: localip.go (hunk 1 above)
Comment (prashantv) [inline localip.go]: Done

**Pull request merged**